R/analyze_binary.R

```diff
@@ -96,32 +87,46 @@ deepstate_fuzz_fun_analyze<- function(test_function,seed=-1, time.limit.seconds,
   runner_harness_name <- paste0(runner_line_split[2], "_", runner_line_split[4])
   
   run.executable <- if(seed == -1 && time.limit.seconds != -1){
-    paste0("cd ",test_function," && valgrind --xml=yes --xml-file=",log_file,
-           " --tool=memcheck --leak-check=yes --track-origins=yes ",
-           "./",basename(test_function),"_DeepState_TestHarness --timeout=",time.limit.seconds,
-           " --fuzz --input_which_test ",runner_harness_name," > ",valgrind.log.text," 2>&1")
+    paste0("cd ",test_function," && valgrind --xml=yes --xml-file=",
+           valgrind_log_xml, " --tool=memcheck --leak-check=yes ",
+           "--track-origins=yes ", "./", basename(test_function),
+           "_DeepState_TestHarness --timeout=", time.limit.seconds, 
+           " --fuzz --input_which_test ", runner_harness_name," > ",
+           valgrind_log_txt," 2>&1")
   }else{
-    paste0("cd ",test_function," && valgrind --xml=yes --xml-file=",log_file,
-           " --tool=memcheck --leak-check=yes --track-origins=yes ",
-           "./",basename(test_function),"_DeepState_TestHarness --seed=",seed,
-           " --timeout=",time.limit.seconds," --fuzz --input_which_test ",runner_harness_name,
-           " > ",valgrind.log.text," 2>&1")
+    paste0("cd ",test_function," && valgrind --xml=yes --xml-file=",
+           valgrind_log_xml, " --tool=memcheck --leak-check=yes ",
+           "--track-origins=yes ", "./",basename(test_function),
+           "_DeepState_TestHarness --seed=",seed, " --timeout=",
+           time.limit.seconds," --fuzz --input_which_test ",
+           runner_harness_name, " > ",valgrind_log_txt," 2>&1")
```

Comment:
@FabrizioSandri can you please refactor this if/else to make is clear that the only difference between the two commands is the --seed arg? Right now there is a lot of duplicated code so the intent is not clear.

Reply:
You are right, I just updated that part an now it is more clear. Solved with commit eb0288e